posixlib/src/main/scala/scala/scalanative/posix/sys/utsname.scala

```diff
@@ -13,10 +13,10 @@ object utsname {
 
 object uname {
   implicit class utsnameOps(val c: Ptr[utsname.utsname]) {
-    def sysname = fromCString(c._1.asInstanceOf[Ptr[CChar]])
-    def nodename = fromCString(c._2.asInstanceOf[Ptr[CChar]])
-    def release = fromCString(c._3.asInstanceOf[Ptr[CChar]])
-    def version = fromCString(c._4.asInstanceOf[Ptr[CChar]])
-    def machine = fromCString(c._5.asInstanceOf[Ptr[CChar]])
+    def sysname = fromCString(c.at1.asInstanceOf[Ptr[CChar]])
+    def nodename = fromCString(c.at2.asInstanceOf[Ptr[CChar]])
+    def release = fromCString(c.at3.asInstanceOf[Ptr[CChar]])
+    def version = fromCString(c.at4.asInstanceOf[Ptr[CChar]])
+    def machine = fromCString(c.at5.asInstanceOf[Ptr[CChar]])
```

Comment:
I would say that the POSIX design here is a bit lacking.  I am not sure about other code that may be similar either off the top of my head. @LeeTibbert, Do you have any ideas?

Reply:
So the point here is whether it is ok or not to implicitly provide Ops that convert to String(where user would normally expect an Array of Char), right?

Reply:
Yeah, in my opinion the point of the Ops is to replace the `_1`, `_2`, ... stuff with the actual struct names for readability. Besides that, it should not do any other conversion—this is supposed to be a raw binding, not a user-friendly interface.

Reply:
@LeeTibbert 
Thank you for the detailed comment.

I also think it's not good to implicitly convert types with ops, so I will change `utsnameOps` to return CArray like other parts.

Of course, I will also update the docs.
I found the 0.5.0 release docs [here](https://github.com/scala-native/scala-native/blob/main/docs/changelog/0.5.0.md) and [build method](https://github.com/scala-native/scala-native/blob/main/docs/contrib/build.rst#locally-publish-docs), so should I summarize the discussion here in this docs and include this PR?

To summarize: I'll make this PR for 0.5.0 instead of 0.4.x, change the behavior of the Ops, and update the docs to reflect this change. Is that okay?


(I apologize in advance for any delays in my response as it's a long holiday week in my country.)

Reply:
re: long holiday weekend

Maybe I should move to your country, at least for the weekend....

Re: PR

Yes, I think the plan you describe is good.   `docs/changelog/0.5.0.md` looks like
the right file.  That is currently a rough draft document so that the wording &
placement need only be a first approximation: fit for purpose and no obvious
blunders/misspellings but not finely tuned wordsmithing.

If the `makedocs` works for you, great.  I have had problems with it; I
think with the installing Sphinx part.  I might assume the presence
of a (python) .mumble file or such which I do not have on my system.

IIRC I have an open Issue here, which I have never gotten the time to fix.

I know sphinx is needed for .rst (rich structured text) files. I do not know
if all of it is needed for .md (markdown) files.

Not to lead anyone astray... I usually change directory to the docs directory
and then run "make html" on the command line. I then use a "file://" URL
with the "//" replaced by  two slashes and the fully qualified (which add a third slash)
name of the _build/html directory (or a specific file under it).

Sorry if this quicky tutorial is describing stuff you already know, perhaps better than I.
Trying to save you time, and importantly frustration/pain.

Contributors are precious; they need encouragement.

Reply:
I’m back :)

@LeeTibbert
Thank you for the detailed explanation, and I was able to build the docs without any issues.

So I modified the function to return a CArray instead of a String, and added some description to 0.5.0 docs about this change.

By the way, I’m wondering if I should fix the behavior of Ops in version 0.4.x.
Should we make a change that only fixes the bug while keeping api to return a string?
(And would that probably be a separate PR?)